BUG: Compilation failure in HiGHs with MSVC

### Describe your issue.

Recently, the following error shows up quite often in CI: https://github.com/scipy/scipy/actions/runs/16271430193/job/45939816354?pr=22580

The error message points to the likely fix already.


### Reproducing Code Example

```python
n/a
```

### Error message

```shell
D:\a\scipy\scipy\subprojects\highs\src\highs_bindings.cpp : fatal error C1128: number of sections exceeded object fileformat limit: compile with /bigobj
```

### SciPy/NumPy/Python version and system information

```shell
main branch
```

--- a/meson.build
+++ b/meson.build
@@ -111,6 +111,10 @@ if cc.get_id() == 'clang-cl'
   add_project_arguments('-Qunused-arguments', language: ['c', 'cpp'])
 endif
 
+if cpp.get_id() == 'msvc'
+  add_project_arguments('/bigobj', language: 'cpp')
+endif
+
 if host_machine.system() == 'darwin'
   if cc.has_link_argument('-Wl,-dead_strip')
     # Allow linker to strip unused symbols

